gobblin-modules/gobblin-http/src/main/java/gobblin/http/ApacheHttpResponseHandler.java

```diff
@@ -18,31 +16,48 @@
  *   sent from the service for a post response, executing more detailed status code handling, etc.
  * </p>
  */
-public class HttpResponseHandler implements ResponseHandler<CloseableHttpResponse> {
-  private static final Logger LOG = LoggerFactory.getLogger(HttpResponseHandler.class);
+@Slf4j
+public class ApacheHttpResponseHandler implements ResponseHandler<CloseableHttpResponse> {
 
   @Override
-  public ResponseStatus handleResponse(CloseableHttpResponse response) {
-    ResponseStatus status = new ResponseStatus(StatusType.OK);
+  public ApacheHttpResponseStatus handleResponse(CloseableHttpResponse response) {
+    ApacheHttpResponseStatus status = new ApacheHttpResponseStatus(StatusType.OK);
     int statusCode = response.getStatusLine().getStatusCode();
-    HttpEntity entity = response.getEntity();
-    if (entity != null) {
-      processEntity(entity);
-    }
-    if (statusCode > 300 & statusCode < 500) {
+    status.setStatusCode(statusCode);
+
+    if (statusCode >= 300 & statusCode < 500) {
       status.setType(StatusType.CLIENT_ERROR);
     } else if (statusCode >= 500) {
       status.setType(StatusType.SERVER_ERROR);
     }
 
+    if (status.getType() == StatusType.OK) {
+      status.setContent(getEntityAsByteArray(response.getEntity()));
+      status.setContentType(response.getEntity().getContentType().getValue());
+    }
+
+    HttpEntity entity = response.getEntity();
+    if (entity != null) {
+      consumeEntity(entity);
```

Comment:
@zxcware I still think consume entity is necessary here because it will close the inputstream. getEntityAsByteArray doesn't consume nor close the inputstream. it's just a query.